### PR TITLE
fail build on docker push failures

### PR DESCRIPTION
**Pull request by mwrock**
Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

File: .buildkite/release_pipeline.yaml [hunk 1]
```diff
@@ -357,7 +357,7 @@ steps:
     env:
       BUILD_PKG_TARGET: "x86_64-linux-kernel2"
 
-  - label: ":docker: :windows: Upload windows container to Bintray"
+  - label: ":docker: :windows: Upload windows container to Docker Hub"
```
Comment (christophermaier) [inline .buildkite/release_pipeline.yaml]: eeeeexcellent

File: .buildkite/scripts/publish_docker_studio.ps1 [hunk 2]
```diff
@@ -21,6 +21,9 @@ Write-Host "--- Publishing the Windows Docker Studio"
 . ./results/last_image.ps1
```
Comment (christophermaier) [inline .buildkite/scripts/publish_docker_studio.ps1]: I can't comment directly, but what if line 18 fails?

File: .buildkite/scripts/publish_docker_studio.ps1 [hunk 3]
```diff
@@ -29,14 +32,24 @@ if(is_fake_release) {
...
     Write-Host "Pushing ${docker_image}:$docker_base_image_tag-$docker_image_short_version tag for $docker_image_version"
     docker push "${docker_image}:$docker_base_image_tag-$docker_image_short_version"
+    if ($LASTEXITCODE -ne 0) {
+      Write-Error "Failed to tag image"
+    }
     if($docker_base_image_tag -eq "ltsc2016") {
       Write-Host "Pushing latest tag for $docker_image_version"
       docker push "${docker_image}:latest"
+      if ($LASTEXITCODE -ne 0) {
+        Write-Error "Failed to tag image"
```
Comment (christophermaier) [inline .buildkite/scripts/publish_docker_studio.ps1]: Would it be worth differentiating this error message from the one on line 41, just to give us a little extra context if something fails?

Outputting which tagging failed would be nice.

Comment (mwrock) [reply .buildkite/scripts/publish_docker_studio.ps1]: I think the `Write-Host` should give that context

File: .buildkite/scripts/publish_docker_studio.ps1 (hunk 2 above)
Comment (mwrock) [reply .buildkite/scripts/publish_docker_studio.ps1]: we have a check on the docker build but although the `docker tag`s are pretty safe, we might as well guard them.

File: .buildkite/scripts/publish_docker_studio.ps1 (hunk 3 above)
Comment (christophermaier) [reply .buildkite/scripts/publish_docker_studio.ps1]: Derp... so it does!

Ignore me :sweat_smile:

**Pull request merged**